lake/root.go

```diff
@@ -1,47 +1,59 @@
 package lake
 
 import (
+	"bytes"
 	"context"
 	"errors"
 	"fmt"
 	"sort"
 
+	"github.com/brimdata/zed/compiler/ast/dag"
+	"github.com/brimdata/zed/expr"
 	"github.com/brimdata/zed/expr/extent"
 	"github.com/brimdata/zed/lake/commit"
 	"github.com/brimdata/zed/lake/index"
+	"github.com/brimdata/zed/lake/journal"
 	"github.com/brimdata/zed/lake/journal/kvs"
 	"github.com/brimdata/zed/lake/segment"
 	"github.com/brimdata/zed/order"
 	"github.com/brimdata/zed/pkg/storage"
 	"github.com/brimdata/zed/proc"
 	"github.com/brimdata/zed/zbuf"
 	"github.com/brimdata/zed/zio"
+	"github.com/brimdata/zed/zngbytes"
 	"github.com/brimdata/zed/zqe"
 	"github.com/brimdata/zed/zson"
 	"github.com/segmentio/ksuid"
 )
 
-var ErrPoolExists = errors.New("pool already exists")
-var ErrPoolNotFound = errors.New("pool not found")
+var (
+	ErrPoolExists   = errors.New("pool already exists")
+	ErrPoolNotFound = errors.New("pool not found")
+)
 
 const (
-	PoolsTag      = "pools"
-	IndexRulesTag = "index_rules"
+	Version         = 1
+	PoolsTag        = "pools"
+	MetaTag         = "metas"
+	IndexRulesTag   = "index_rules"
+	LakeMagicFile   = "lake.zng"
+	LakeMagicString = "ZED LAKE"
 )
 
 // The Root of the lake represents the path prefix and configuration state
 // for all of the data pools in the lake.
 type Root struct {
-	*Config
-	engine storage.Engine
-	path   *storage.URI
+	engine     storage.Engine
+	path       *storage.URI
+	pools      *kvs.Store
+	indexRules *index.Store
```

Comment:
```suggestion
	indexStore *index.Store
```

Reply:
it's the store of the rules.  package index manages the rules.  maybe better to change package index to package indexrules?